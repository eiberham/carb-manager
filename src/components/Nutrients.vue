<template>
  <section class="nutrients">
    <div class="nutrients__carbs">{{ nutrients.carbs }} g</div>
    <div class="nutrients__proteins">{{ nutrients.proteins }} g</div>
    <div class="nutrients__fats">{{ nutrients.fats }} g</div>
  </section>
</template>

<script>
export default {
  props: {
    nutrients: {
      type: Object,
      required: true
    }
  }
};
</script>

<style scoped>
* {
  box-sizing: border-box;
}

.nutrients {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
}

.nutrients > * {
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 14px;
  color: #6f737a;
  position: relative;
  padding-left: 8px;
  padding-top: 8px;
}

.nutrients > *::before {
  content: "";
  display: block;
  width: 6px;
  height: 6px;
  position: absolute;
  top: 12px;
  left: 0;
  border-radius: 50%;
}

.nutrients__carbs::before {
  background: #f94642;
}

.nutrients__proteins::before {
  background: #3177bb;
}

.nutrients__fats::before {
  background: #fda120;
}
</style>
