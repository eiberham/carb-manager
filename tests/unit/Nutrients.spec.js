import { mount } from "@vue/test-utils";
import Nutrients from "../../src/components/Nutrients.vue";

describe("Nutrients.vue", () => {
  const nutrients = {
    proteins: 20,
    carbs: 16,
    fats: 6
  };

  const wrapper = mount(Nutrients, {
    propsData: {
      nutrients
    }
  });

  test("is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  test("received props", () => {
    expect(wrapper.find(".nutrients__proteins").text()).toContain(
      nutrients.proteins
    );
    expect(wrapper.find(".nutrients__carbs").text()).toContain(nutrients.carbs);
    expect(wrapper.find(".nutrients__fats").text()).toContain(nutrients.fats);
  });

  test("matches snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
