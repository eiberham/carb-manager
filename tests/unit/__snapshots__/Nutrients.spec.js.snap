// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nutrients.vue matches snapshot 1`] = `
<section
  class="nutrients"
>
  <div
    class="nutrients__carbs"
  >
    16 g
  </div>
   
  <div
    class="nutrients__proteins"
  >
    20 g
  </div>
   
  <div
    class="nutrients__fats"
  >
    6 g
  </div>
</section>
`;
